Fix: OSW, pass im_range to scoreIdentification method

singjc opened the pull request:
## Description

This addresses issue #7883 

I now pass the im_range to `scoreIdentification_` instead of recomputing it within the function, which results in an im_range with (-1,-1) if there is no drift time set in the library (i.e. non-ion mobility data).

Not sure why it was initially done that way, I'm assuming we wouldn't want to recompute it in the scoreIdentifiation method since these traces will have have the same library drift time as the detecting transitions.

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced optional ion mobility filtering for a more refined scoring process.
- **Documentation**
  - Updated user guidance to explain the new ion mobility parameter for improved analysis control.
- **Bug Fixes**
  - Resolved issue with `im_range` handling by requiring it to be passed directly to the scoring method.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

singjc committed:
fix: pass ion mobility range to scoreIdentification, instead of it re-initializing it within the method

singjc committed:
chore: update CHANGELOG

timosachsenberg committed:
Update CHANGELOG

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

timosachsenberg commented on CHANGELOG:
```diff
@@ -27,6 +27,8 @@ OpenSwath:
 - Add peak-picking for extracted ion mobilogram (#7759)
 - Add ion mobility scoring for identifying transitions for IPF (#7760)
+- - Bug fix, pass `im_range` to `scoreIdentifiction_` instead of computing it in the fucntion. Addresses issue #7883. (PR #7885)
++ - Bug fix, pass `im_range` to `scoreIdentification_` instead of computing it in the function. Addresses issue #7883. (PR #7885)
```
```suggestion
- Bug fix, pass `im_range` to `scoreIdentification_` instead of computing it in the function. Addresses issue #7883. (PR #7885)
```

timosachsenberg committed:
Update CHANGELOG

timosachsenberg commented on src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.h:
```diff
@@ -243,6 +243,7 @@ namespace OpenMS
      *                  the chromatograms were extracted. Use empty map if no
      *                  data is available.
      * @param drift_target The target drift value
+     * @param im_range Ion mobility subrange to consider (used as filter); can be empty (i.e. no IM filtering)
```
Does this require some python binding updates?
Maybe this can provide more information. What happens if non IM data is scored and a user sets im_range? What if the user has IM data but the range is empty?
Should there be a default (e.g. the empty range?)

singjc replied:
No python binding updates required, this method is not wrapped.
I updated the doc to clarify that it should be empty for non-IM data.  If the use does have IM data, and it's empty, then we assume the user does not want to use the ion mobility information. The IM range is set from the library, so we assume/depend that the library was generated with IM information coordinates.

jcharkow reviewed (approved):
Looks good to me!

singjc committed:
chore: update scoreIdentification_ doc

- Add more information to clarify when scoring non-IMS data, it should be empty.

singjc committed:
Merge branch 'fix/osw_scoreident_imrange' of github.com:singjc/OpenMS into fix/osw_scoreident_imrange

singjc committed:
Merge branch 'develop' into fix/osw_scoreident_imrange
